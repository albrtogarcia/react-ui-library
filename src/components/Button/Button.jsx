import PropTypes from 'prop-types'
import './Button.scss'

const Button = ({ type, children, handleClick }) => (
  <button className={`btn btn-${type}`} onClick={handleClick}>
    {children}
  </button>
)

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node.isRequired,
  handleClick: PropTypes.func,
}

export default Button
