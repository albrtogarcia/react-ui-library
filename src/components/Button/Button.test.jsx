import { describe, expect, it } from 'vitest'
import Button from './Button'

describe('Button', () => {
  const props = {
    type: 'primary',
    children: 'Click Me',
    handleClick: () => {},
  }

  it('matches snapshot', () => {
    const button = Button(props)
    expect(button).toMatchSnapshot()
  })
})
