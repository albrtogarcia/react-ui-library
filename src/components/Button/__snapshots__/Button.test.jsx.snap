// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > matches snapshot 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
>
  Click Me
</button>
`;
